src/main/java/com/grubBuddy/ingest/service/FoodService.java

```diff
@@ -25,23 +33,55 @@ public void syncFoodsDatabase() throws InterruptedException {
         while(!isDone[0]) {
             if(inProcess[0] != 5) {
                 inProcess[0] = inProcess[0] + 1;
-                this.foodDAO.insertMany(this.foodListApi.getFoods(take, page + 1)
-                    .map(x -> {
-                        if(x == null) {
-                            isDone[0] = true;
-                            return null;
-                        };
-                        return x.list.foodItems;
-                    }))
-                    .doOnComplete(() -> inProcess[0] = inProcess[0] - 1)
-                    .doOnError(err -> isDone[0] = true)
-                    .subscribeOn(Schedulers.parallel())
-                    .subscribe();
-
+                syncFoodAndNutrients(page, take, isDone, inProcess);
                 page += take;
             }
+
             Thread.sleep(0, 1);
         }
     }
 
+    private void syncFoodAndNutrients(int page, int take, boolean[] isDone, int[] inProcess) {
+        Mono<Tuple2<FoodApiResponse, FoodApiResponse>> dataSync = foodItemSync(page, take);
```

Comment:
Why a tuple? If we have to add another item (e.g. food group), it'll be more work.

Reply:
Because thats what .zip returns in the foodItemSync method. If you need to add another item just change it to mergeWith.